packages/material-ui/src/TablePagination/TablePagination.js

```diff
@@ -1,74 +1,144 @@
-import TablePaginationActions from './TablePaginationActions';
+import PaginationActions from './TablePaginationActions';
 import useId from '../utils/useId';
+import { getTablePaginationUtilityClass } from './tablePaginationClasses';
+
+const makeOverridesResolver = (slotName) => {
+  return function overridesResolver(_, styles) {
+    return styles[slotName];
+  };
+};
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    color: theme.palette.text.primary,
-    fontSize: theme.typography.pxToRem(14),
-    overflow: 'auto',
-    // Increase the specificity to override TableCell.
-    '&:last-child': {
-      padding: 0,
-    },
+const TablePaginationRoot = experimentalStyled(
+  TableCell,
+  { shouldForwardProp: (prop) => shouldForwardProp(prop) || prop === 'classes' },
+  { name: 'MuiTablePagination', slot: 'Root', overridesResolver: makeOverridesResolver('root') },
+)(({ theme }) => ({
+  color: theme.palette.text.primary,
+  fontSize: theme.typography.pxToRem(14),
+  overflow: 'auto',
+  // Increase the specificity to override TableCell.
+  '&:last-child': {
+    padding: 0,
   },
-  /* Styles applied to the Toolbar component. */
-  toolbar: {
+}));
+
+const TablePaginationToolbar = experimentalStyled(
+  Toolbar,
+  { shouldForwardProp: (prop) => shouldForwardProp(prop) || prop === 'classes' },
+  {
+    name: 'MuiTablePagination',
+    slot: 'Toolbar',
+    overridesResolver: makeOverridesResolver('toolbar'),
+  },
+)(({ theme }) => ({
+  minHeight: 52,
+  paddingRight: 2,
+  [`${theme.breakpoints.up('xs')} and (orientation: landscape)`]: {
     minHeight: 52,
-    paddingRight: 2,
   },
-  /* Styles applied to the spacer element. */
-  spacer: {
-    flex: '1 1 100%',
+  [theme.breakpoints.up('sm')]: {
+    minHeight: 52,
+    paddingRight: 2,
   },
-  /* Styles applied to the select label Typography element. */
-  selectLabel: {
-    flexShrink: 0,
+}));
+
+const Spacer = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Spacer',
+    overridesResolver: makeOverridesResolver('spacer'),
   },
-  // TODO v5: `.selectRoot` should be merged with `.input`
-  /* Styles applied to the Select component root element. */
-  selectRoot: {
-    marginRight: 32,
-    marginLeft: 8,
+)({
+  flex: '1 1 100%',
+});
+
+const SelectLabel = experimentalStyled(
+  Typography,
+  { shouldForwardProp: (prop) => shouldForwardProp(prop) || prop === 'classes' },
+  {
+    name: 'MuiTablePagination',
+    slot: 'SelectLabel',
+    overridesResolver: makeOverridesResolver('selectLabel'),
   },
-  /* Styles applied to the Select component `select` class. */
-  select: {
-    paddingLeft: 8,
-    paddingRight: 24,
-    textAlign: 'right',
-    textAlignLast: 'right', // Align <select> on Chrome.
+)({
+  flexShrink: 0,
+});
+
+const TablePaginationSelect = experimentalStyled(
+  Select,
+  { shouldForwardProp: (prop) => shouldForwardProp(prop) || prop === 'classes' },
```

Comment:
It shouldn't be needed since #25780

```suggestion
  {},
```